develop/app/Controllers/v2/PaymentController.php

```diff
@@ -0,0 +1,250 @@
+            "o_key"  => $paymentEntity->o_key,
+            "status" => $paymentEntity->status,
+            "total"  => $paymentEntity->total
+        ];
+
+        return $this->respond([
+            "status" => true,
+            "data"   => $data,
+            "msg"    => "Payment show method successful",
+        ]);
+    }
+
+    /**
+     * [POST] /api/v1/payment
+     * Create payment and reduce user balance.
+     *
+     * @return void
+     */
+    public function create()
+    {
+        $data   = $this->request->getJSON(true);
+        $u_key  = $this->u_key;
+        $o_key  = $data["o_key"];
+        $amount = $data["amount"] ?? null;
+        $price  = $data["price"]  ?? null;
+        $status = "paymentCreate";
+
+        if (is_null($u_key) || is_null($o_key) || is_null($amount) || is_null($price)) {
+            return $this->fail("Incoming data not true", 400);
+        }
+
+        $total = $amount * $price;
+
+        $paymentModel  = new PaymentModel();
+        $walletModel   = new WalletModel();
+
+        $paymentEntity = $paymentModel->where("u_key", $u_key)
+                                      ->where("o_key", $o_key)
+                                      ->first();
+        if (!is_null($paymentEntity)) {
+            return $this->fail("This payment information is not exist or cannot found.", 403);
+        }
+
+        $userWallet = $walletModel->where('u_key', $u_key)
+                                  ->first();
+        if (is_null($userWallet)) {
+            return $this->fail("This user isn't exist.", 400);
+        }
+
+        $userBalance = $userWallet->balance;
+
+        if ($userBalance < $total) {
+            return $this->fail("Insufficient balance", 400);
+        }
+
+        $paymentCreatedIDOrNull = $paymentModel->createPaymentTransaction($u_key, $o_key, $total, $userBalance, $status);
+        if (is_null($paymentCreatedIDOrNull)) {
+            return $this->fail("Payment created failed.", 400);
+        }
+
+        return $this->respond([
+            "status"    => true,
+            "paymentID" => $paymentCreatedIDOrNull,
+            "msg"       => "Payment create method successful."
+        ]);
+    }
+
+    /**
+     * [PUT] /api/v1/payment
+     * Update payment price.
+     *
+     * @param int $paymentKey
+     * @return void
+     */
+    public function update($paymentKey = null)
+    {
+        $data = $this->request->getJSON(true);
+
+        if (is_null($paymentKey)) {
+            return $this->fail("The payment key is required.", 404);
+        }
+
+        if (is_null($data["total"]) || is_null($paymentKey)) {
+            return $this->fail("Incoming data not true", 400);
+        }
+
+        $total  = $data["total"];
+        $status = $data["status"] ?? "paymentUpdate";
+
+        $paymentModel  = new PaymentModel();
+
+        $paymentEntity = $paymentModel->where("u_key", $this->u_key)
+                                      ->find($paymentKey);
```

Comment:
Same. This logic should be extracted, the reusability is too high.

Reply:
This change is done.